From f66912b57323550e0ea87f7e2a487b42f124c9d1 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Wed, 6 Jan 2016 11:06:45 +0800
Subject: [PATCH 0816/5242] MLK-12126 ARM: dts: imx6/imx7d: correct wdog
 pinctrl setting

commit  d169d3dcbc712739a25325fc05f8b5209f25419f from
https://source.codeaurora.org/external/imx/linux-imx.git

We move WDOG_B pinctrl setting from pinctrl_hog to wdog driver, so need to
remove the original setting in pinctl_hog device node, otherwise the below
warning message will be caught:

[    1.284161] imx6ul-pinctrl 20e0000.iomuxc: pin MX6UL_PAD_LCD_RESET already requested by 20e0000.iomuxc; cannot claim for 20bc000.wdog
[    1.294990] imx6ul-pinctrl 20e0000.iomuxc: pin-69 (20bc000.wdog) status -22
[    1.300689] imx6ul-pinctrl 20e0000.iomuxc: could not request pin 69 (MX6UL_PAD_LCD_RESET) from group wdoggrp  on device 20e0000.iomuxc
[    1.311550] imx2-wdt 20bc000.wdog: Error applying setting, reverse things back

Signed-off-by: Robin Gong <b38343@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6qdl-sabresd.dtsi        |    1 -
 arch/arm/boot/dts/imx6ul-14x14-evk.dtsi       |    1 -
 arch/arm/boot/dts/imx6ul-9x9-evk.dts          |   13 ++++++++++++-
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts |   12 ++++++++++++
 4 files changed, 24 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
index 1989c3a..03e7959 100644
--- a/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
+++ b/arch/arm/boot/dts/imx6qdl-sabresd.dtsi
@@ -690,7 +690,6 @@
 				MX6QDL_PAD_SD3_RST__GPIO7_IO08	0x80000000
 				MX6QDL_PAD_GPIO_9__GPIO1_IO09 	0x80000000
 				MX6QDL_PAD_EIM_DA9__GPIO3_IO09 0x80000000
-				MX6QDL_PAD_GPIO_1__WDOG2_B 0x80000000
 				MX6QDL_PAD_NANDF_CS0__GPIO6_IO11 0x80000000
 				MX6QDL_PAD_NANDF_CS1__GPIO6_IO14 0x80000000
 				MX6QDL_PAD_NANDF_CS2__GPIO6_IO15 0x80000000
diff --git a/arch/arm/boot/dts/imx6ul-14x14-evk.dtsi b/arch/arm/boot/dts/imx6ul-14x14-evk.dtsi
index 17f0da2..c8b8cc2 100644
--- a/arch/arm/boot/dts/imx6ul-14x14-evk.dtsi
+++ b/arch/arm/boot/dts/imx6ul-14x14-evk.dtsi
@@ -422,7 +422,6 @@
 
 	pinctrl_hog_1: hoggrp-1 {
 		fsl,pins = <
-			MX6UL_PAD_LCD_RESET__WDOG1_WDOG_ANY    0x30b0
 			MX6UL_PAD_UART1_RTS_B__GPIO1_IO19	0x17059 /* SD1 CD */
 			MX6UL_PAD_GPIO1_IO05__USDHC1_VSELECT	0x17059 /* SD1 VSELECT */
 			MX6UL_PAD_GPIO1_IO09__GPIO1_IO09        0x17059 /* SD1 RESET */
diff --git a/arch/arm/boot/dts/imx6ul-9x9-evk.dts b/arch/arm/boot/dts/imx6ul-9x9-evk.dts
index 00623b0..add35b8 100644
--- a/arch/arm/boot/dts/imx6ul-9x9-evk.dts
+++ b/arch/arm/boot/dts/imx6ul-9x9-evk.dts
@@ -444,7 +444,6 @@
 
 		pinctrl_hog_1: hoggrp-1 {
 			fsl,pins = <
-				MX6UL_PAD_LCD_RESET__WDOG1_WDOG_ANY    0x30b0
 				MX6UL_PAD_UART1_RTS_B__GPIO1_IO19	0x17059 /* SD1 CD */
 				MX6UL_PAD_GPIO1_IO05__USDHC1_VSELECT	0x17059 /* SD1 VSELECT */
 				MX6UL_PAD_GPIO1_IO09__GPIO1_IO09        0x17059 /* SD1 RESET */
@@ -630,6 +629,12 @@
 				MX6UL_PAD_NAND_DATA03__USDHC2_DATA3 0x17059
 			>;
 		};
+
+		pinctrl_wdog: wdoggrp {
+			fsl,pins = <
+				MX6UL_PAD_LCD_RESET__WDOG1_WDOG_ANY    0x30b0
+			>;
+		};
 	};
 };
 
@@ -776,3 +781,9 @@
 	enable-sdio-wakeup;
 	status = "okay";
 };
+
+&wdog1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_wdog>;
+	fsl,ext-reset-output;
+};
diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
index fbc6839..e1732a8 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
@@ -840,6 +840,12 @@
 				MX7D_PAD_LPSR_GPIO1_IO07__GPIO1_IO7    0x14
 			>;
 		};
+
+		pinctrl_wdog: wdoggrp {
+			fsl,pins = <
+				MX7D_PAD_LPSR_GPIO1_IO00__WDOG1_WDOG_B 0x74
+			>;
+		};
 	};
 };
 
@@ -990,3 +996,9 @@
 	pinctrl-0 = <&pinctrl_sim1_1>;
 	status = "okay";
 };
+
+&wdog1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_wdog>;
+	fsl,ext-reset-output;
+};
-- 
1.7.9.5

